Move OCR Combine code to etl pipeline

priya-pai opened the pull request:
## Describe your changes
This PR moves the code related to the OCR combination and tagging for PDFs from the pdf-pipeline repo (https://github.com/NYPL/pdf-pipeline) to this repo, so that we can set up ECS tasks and the other relevant infrastructure. There might be some further changes needed, especially in how the process is initialized and runs. Right now, it parses an event sent by a step function, but we'll want to change it to parse messages sent by a SQS Queue. For now, we've moved almost everything over, except the s3.py - for that one, I just added the one missing method to our existing s3Manager.

Once we have more clarity on the format of messages sent (SQS, Step Functions, etc.), I'll create a follow up PR that copies/adds tests for these files.

## How to test
Currently, this process won't easily be able to be tested until we set up the necessary infrastructure. We want to mainly check that the code has been migrated over correctly and we have the necessary files + processes set up in order to set up the infrastructure.

priya-pai commented on etl-pipeline/processes/util/marc.py:
```diff
@@ -0,0 +1,150 @@
+
+from lxml import etree
+
+
```
@kylevillegas93 I added these lib files to the `util` folder but let me know if you think there's a better spot for them!

kylevillegas93 commented on etl-pipeline/requirements.txt:
```diff
@@ -15,6 +15,7 @@ google-auth-httplib2==0.2.0
 python-gnupg==0.5.0
```
nope i don't think so - just these: https://github.com/NYPL/pdf-pipeline/blob/main/src/ocr_combiner/requirements.txt

priya-pai replied:
Got it! https://github.com/NYPL/digital-research-books/pull/778/commits/69428a7a01cf099037f367ad67873390da01000f

kylevillegas93 commented on etl-pipeline/processes/util/pdf_util.py:
```diff
@@ -0,0 +1,25 @@
+from typing import Iterator
+
+
+def get_logger(name: str) -> logging.Logger:
```
I think we can use the etl-pipeline logger!

priya-pai replied:
good callout! fixed here - https://github.com/NYPL/digital-research-books/pull/778/commits/69428a7a01cf099037f367ad67873390da01000f

kylevillegas93 commented:
For Folder structure - recommend creating a folder for pdf_generation and can add the process and utils folder there!

kylevillegas93 commented on etl-pipeline/managers/s3.py:
```diff
@@ -97,6 +99,23 @@ def generate_manifest(self, record: Record, source_url: str, manifest_url: str):
+    def upload_file(
+        self,
+        file: BytesIO,
+        key: typing.TypeAlias = str | pathlib.Path,
```
hmm does it need TypeAlias here or can it be `key: str | pathlib.Path`

kylevillegas93 commented on etl-pipeline/processes/util/pdf_page.py:
```diff
@@ -0,0 +1,173 @@
+
+import checksum
+import mets_parser
+import s3
```
I think the s3 manager in the PDF pipeline is a bit different so we might have to port that over as well

priya-pai commented:
@kylevillegas93 Addressed all of your comments and I created a folder within processes called `pdf_generation` and added the relevant lib files there! But let me know if you meant create a `pdf_generation` folder at the highest level in `etl-pipeline`

kylevillegas93 commented on etl-pipeline/processes/grin/conversion.py:
```diff
@@ -22,12 +22,13 @@ def __init__(self, *args, batch_limit=1000):
+            # self.acquire_and_convert_new_books()
 
-            self.process_converted_books()
+            # self.process_converted_books()
 
-            if backfill:
-                self.convert_backfills()
+            # if backfill:
+            #     self.convert_backfills()
```
should we uncomment these?

kylevillegas93 commented on etl-pipeline/processes/pdf_generation/path.py:
```diff
@@ -0,0 +1,74 @@
+import pathlib
+import re
+
+COMBINED_OCR_PDF_DIR = pathlib.Path("combined_ocr_pdfs")
+METADATA_DIR = pathlib.Path("metadata_files")
+OCR_INPUT_DIR = pathlib.Path("ocr_inputs")
+TAGGED_PDF_DIR = pathlib.Path("tagged_pdfs")
+UNZIPPED_OCR_DIR = pathlib.Path("unzipped_ocr")
```
nice :)

kylevillegas93 commented on etl-pipeline/processes/pdf_generation/pdf_generate.py:
```diff
@@ -0,0 +1,127 @@
+
+
+class PDFGenerationProcess:
+    def __init__(self, event):
```
Looks good - we can circle back to think more about the params for this process
